lib/FrontendTool/FrontendTool.cpp

```diff
@@ -219,38 +239,79 @@ static bool emitLoadedModuleTraceIfNeeded(ASTContext &ctxt,
     return true;
   }
 
-  llvm::SmallVector<std::string, 16> swiftModules;
+  ModuleDecl* currentModule = ctxt.getModuleByName(moduleName);
+  ModuleDecl::ImportFilter filter = ModuleDecl::ImportFilterKind::Public;
+  filter |= ModuleDecl::ImportFilterKind::Private;
+  filter |= ModuleDecl::ImportFilterKind::ImplementationOnly;
+  llvm::SmallVector<ModuleDecl::ImportedModule, 8> imports;
```

Comment:
We adopt a number of common LLVM helper types into the `swift` namespace to save on line noise; you can see these in `swift/Basic/LLVM.h`.